packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -36,18 +36,42 @@ enum _ScaffoldSlot {
   statusBar,
 }
 
+/// Geometry information for scaffold components.
+///
+/// To get a [ValueNotifier] for the scaffold gemoetry call
+/// [Scaffold.geometryOf].
+@immutable
+class ScaffoldGeometry {
+  /// The rectangle in which the scaffold is laying out
+  /// [Scaffold.bottomNavigationBar].
+  /// Null when there is no bottom navigation bar..
+  final Rect bottomNavigationBar;
+
+  /// The rectangle in which the scaffold is laying out
+  /// [Scaffold.floatingActionButton].
+  /// Null when there is no bottom navigation bar..
+  final Rect floatingActionButton;
+
+  const ScaffoldGeometry({
```

Comment:
nit: constructors first (when first reading this i assumed there was an implicit constructor, but when i got to this constructor i realised no, actually, there's no implicit constructor)

Reply:
done